colortoypiano/colortoypiano.py

```diff
@@ -0,0 +1,84 @@
+"""
+The colortoypiano uses multi-threading to interface with
+a chosen midi interface. This interface is used to print all
+incoming midi messages for that device.
+"""
+
+import colortoypianoshell
+import mido
+import queue
+import sys
+import threading
+
+
+class ColorToyPiano(object):
+
+    def __init__(self):
+        """Create local queue and thread
+
+        Args:
+            self (ColorToyPiano object): a reference to self.
+        """
+        self.q = queue.Queue()
+        self.t = None
+
+    def quit(self):
+        """Quit child threads
+
+        Args:
+            self (ColorToyPiano object): a reference to self.
+        """
+        if self.t is not None:
+            # Pass 'quit' command to child thread.
+            self.q.put('quit')
+            while self.t.is_alive():
```

Comment:
https://docs.python.org/3/library/threading.html#threading.Thread.join

Docs say it only raises an exception if the thread hasn't started yet.  If it has already terminated, join() will just return immediately.